### DISCUSSION TITLE
Uvicorn sending invalid responses when started with uvicorn.run

### DISCUSSION THREAD

**Discussion by jrhe** [Kludex/uvicorn#1949] [Potential Issue]
Uvicorn returns invalid responses when run with `uvicorn.run(app, host="0.0.0.0", port=9000)` in a python file `main.py`, but works when I run it from the command line with `uvicorn main:app`.

The invalid response looks like the following, which is clearly truncated:
```
ÿ�������;�Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/112.�
Accept: text/html,application/xhtml+xml,applcation/xml;q=0.9,image/avif,image/webp,*/*;q=0.8
Accept-Language: en-GB,en;q=0.5
Accept-Encoding: gzip, de�late, br
DNT: 1
Connection: keep-alive
Upgrade-Insecure-Requests: 1
Sec-Fetch-Dest: document
Se
```

I thought it might have been an issue with corrupt dependencies so I cleared all module caches (pip, pyenv, poetry etc), blew away the virtualenv and recreated it, tried updating my lockfile, etc, with no avail.

I have recreated it with a minimal example as follows

```
async def app(scope, receive, send):
    assert scope['type'] == 'http'

    await send({
        'type': 'http.response.start',
        'status': 200,
        'headers': [
            [b'content-type', b'text/plain'],
        ],
    })
    await send({
        'type': 'http.response.body',
        'body': b'Hello, world!',
    })



if __name__ == "__main__":
    import uvicorn

    uvicorn.run(app, host="0.0.0.0", port=9000)
```

Any pointers to debug very much appreciated as I'm at a loss as to how to resolve it.

**Comment by parthvnp** [comment]
@jrhe I will investigate this and let you know.
